rules/TypeDeclaration/Rector/Property/TypedPropertyFromAssignsRector.php

```diff
@@ -154,6 +155,15 @@ public function refactor(Node $node): ?Node
         return $node;
     }
 
+    private function processChangedPhpDocInfo(PhpDocInfo $phpDocInfo, Node $node): ?Node
+    {
+        if ($phpDocInfo->hasChanged()) {
+            return $node;
+        }
+
+        return null;
+    }
```

Comment:
What is this for?

This rule should only change the strict type declarations and avoid docblocks. 
Also this can report false positive change in case of another node being changed incorrectly.

Reply:
it called when only change docblock, see in line 131, no type changed, only change docblock, then check PhpDocInfo is changed.

Reply:
We currently stil mark `DateTime|null` as union type in `UnionTypeMapper`, not `Nullable`, so then we use docblock in PHP 7.4 feature set.

That need separate PR to make it nullable compatible `DateTime|null` marked as nullable type in PHP 7.4, ref https://3v4l.org/4HTQT#v7.4.33

Reply:
I added new fixture for php 8.x feature set to make it strict typed https://github.com/rectorphp/rector-src/pull/3161/files#diff-1dfc0caeebe902ff508cdf3b3bc6d0a09ea00526c8c7ff572f75bbd0c51f4a77

Reply:
I added another fixture for php 8.x feature enabled to remove existing docblock https://github.com/rectorphp/rector-src/pull/3161/commits/d12ba31af2172a79fc55d8bdd852e8fe77330649

- The default test is using php 7.4 feature, so it uses docblock ( union nullable still marked as union, not nullable)
- when we use php 8.x feature on `FixtureComplexTypes` directory, it uses fully php 8.x feature.

Reply:
The goal is to avoid changing docblocks in anyway, as they can be used later as relly on strict-type declarations. 
Instead the code on PHP 7.4- for union types should be skipped, and strict union type added only in PHP 8.0.

That way we always add types that are known and strict :+1: